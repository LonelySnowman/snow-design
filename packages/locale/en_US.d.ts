import type { Locale } from './index';
declare const localeValues: Locale;
export default localeValues;
