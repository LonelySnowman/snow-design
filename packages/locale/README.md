# @snow-design/locale

SnowDesign 组件库国际化语言包。
