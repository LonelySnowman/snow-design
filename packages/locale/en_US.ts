import type { Locale } from './index';

const localeValues: Locale = {
    locale: 'en',
    Pagination: {
        pageSize: 'Items per page: ${pageSize}',
        total: 'Total pages: ${total}',
        previous: 'Previous',
        next: 'Next',
        jumpTo: 'Jump to',
        page: ' page',
    },
};

export default localeValues;
