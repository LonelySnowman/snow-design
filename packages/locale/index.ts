export interface Locale {
  locale: string;
  Pagination: {
    pageSize: string;
    total: string;
    previous: string,
    next: string,
    jumpTo: string;
    page: string;
  }
}

export type LocaleComponentName = Exclude<keyof Locale, 'locale'>;
