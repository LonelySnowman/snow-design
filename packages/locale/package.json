{
  "name": "@snow-design/locale",
  "version": "0.0.22",
  "files": [
    "lib"
  ],
  "scripts": {
    "build": "tsup",
    "clean": "rimraf lib node_modules"
  },
  "description": "SnowDesign 组件库国际化语言包",
  "main": "index.ts",
  "keywords": ["snow-design"],
  "author": "LonelySnowman",
  "license": "ISC",
  "devDependencies": {
    "tsup": "^8.2.3"
  }
}
