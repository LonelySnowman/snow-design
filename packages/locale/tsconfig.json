{
  "compilerOptions": {
    "target": "ES2015",
    "esModuleInterop": true,
    "outDir": "lib",
    "sourceMap": true,
    "allowJs": true,
    "module": "es6",
    "moduleResolution": "node",
    "lib": ["es7", "dom"],
    "jsx": "react",
    "skipLibCheck": true,
    "declaration": true,
    "strictNullChecks": false,
    "paths": {
      "@snow-design/foundation/*": ["../foundation/*"],
      "@snow-design/components/*": ["../components/*"],
    }
  },
  "include": ["**/*.tsx", "**/*.ts"],
  "exclude": ["node_modules"]
}
