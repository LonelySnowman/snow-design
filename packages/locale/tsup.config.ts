import { defineConfig } from 'tsup';

const baseConfig = {
    entry: ['index.ts', 'en_US.ts', 'zh_CN.ts'],
    clean: true,
    outExtension() {
        return {
            js: '.js',
            dts: '.ts',
        };
    },
};

export default defineConfig(() => {
    return [
        {
            ...baseConfig,
            format: 'esm',
            outDir: 'lib/es',
        },
        {
            ...baseConfig,
            format: 'cjs',
            outDir: 'lib/cjs',
            dts: true,
        },
    ];
});
