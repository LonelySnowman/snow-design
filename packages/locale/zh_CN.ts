import type { Locale } from './index';

const localeValues: Locale = {
    locale: 'en',
    Pagination: {
        pageSize: '每页大小: ${pageSize}',
        total: '总页数: ${total}',
        previous: '上一页',
        next: '下一页',
        jumpTo: '前往',
        page: '页',
    },
};

export default localeValues;
