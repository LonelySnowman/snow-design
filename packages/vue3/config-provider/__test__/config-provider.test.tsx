import mountTest from '#test/vue/mountTest';
import '@testing-library/jest-dom';
import ConfigProvider from '../index';
import Pagination from '@snow-design/vue3/pagination';
import { render } from '@testing-library/vue';
import { zh_CN } from '@snow-design/locale';

describe('ConfigProvider', () => {
    mountTest(<ConfigProvider />);

    it('renders correctly', () => {
        const { container } = render(
            <ConfigProvider>
                <Pagination showTotal total={20} pageSize={2}></Pagination>
            </ConfigProvider>,
        );
        expect(container).toMatchSnapshot();
    });

    it('use the correct language', () => {
        expect(
            render(
                <ConfigProvider locale={zh_CN}>
                    <Pagination showTotal total={20} pageSize={2}></Pagination>
                </ConfigProvider>,
            ).container,
        ).toContainHTML('总页数');
        expect(
            render(
                <ConfigProvider>
                    <Pagination showTotal total={20} pageSize={2}></Pagination>
                </ConfigProvider>,
            ).container,
        ).toContainHTML('Total pages');
    });
});
