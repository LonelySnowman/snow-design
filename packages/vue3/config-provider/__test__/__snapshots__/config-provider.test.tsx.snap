// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigProvider renders correctly 1`] = `
<div>
  
  
  <ul
    class="snow-pagination"
  >
    <span
      class="snow-pagination-total"
    >
      Total pages: 10
    </span>
    <li
      aria-disabled="true"
      aria-label="Previous"
      class="snow-pagination-item snow-pagination-prev snow-pagination-item-disabled"
      role="button"
    >
      Previous
    </li>
    
    <li
      aria-current="page"
      aria-label="Page 1"
      class="snow-pagination-item snow-pagination-item-active"
    >
      1
    </li>
    <li
      aria-current="false"
      aria-label="Page 2"
      class="snow-pagination-item"
    >
      2
    </li>
    <li
      aria-current="false"
      aria-label="Page 3"
      class="snow-pagination-item"
    >
      3
    </li>
    <li
      aria-current="false"
      aria-label="Page 4"
      class="snow-pagination-item"
    >
      4
    </li>
    <li
      aria-current="false"
      aria-label="More"
      class="snow-pagination-item"
    >
      ...
    </li>
    <li
      aria-current="false"
      aria-label="Page 9"
      class="snow-pagination-item"
    >
      9
    </li>
    <li
      aria-current="false"
      aria-label="Page 10"
      class="snow-pagination-item"
    >
      10
    </li>
    
    <li
      aria-disabled="false"
      aria-label="Next"
      class="snow-pagination-item snow-pagination-next"
      role="button"
    >
      Next
    </li>
  </ul>
  
  
</div>
`;
